tests/rp_cloud_cleanup.py

```diff
@@ -418,13 +421,41 @@ def _delete_nat(self, nat):
 
         return r
 
-    def clean_aws_nat(self):
+    def _delete_vpc(self, vpc):
+        # Delete VPC
+        # self.log.info(f"-> deleting VPC: {vpc['VpcId']}")
+        _start = time.time()
+
+        # - terminate all instances running in the VPC
+        # - delete all security groups associated with the VPC (except the default one)
+        # - delete all route tables associated with the VPC (except the default one)
+
+        r = self.provider.delete_vpc(vpc['VpcId'], clean_dependencies=True)
+        if not r:
+            self.log.info(f"-> VPC {vpc['VpcId']} NOT deleted")
+            return r
+        _now = time.time()
+        # 180 sec timeout
+        while (_now - _start) < 180:
+            try:
+                r = self.provider.get_vpc_by_id(vpc['VpcId'])
+                if r['State'] == 'deleted':
+                    break
+            except:
+                break
+            time.sleep(10)
+        _elapsed = time.time() - _start
+        self.log.info(f"-> VPC {vpc['VpcId']} deleted, {_elapsed:.2f}s")
```

Comment:
This message isn't neccessarily correct right? Because we may have timed out? We should log that case differently.

BTW this whole loop could be replaced by `ducktape.utils.wait_until` I believe, passing in your predicate. This also allows you to handle the timeout case differently (since it throws).

Reply:
I am trying to eliminate unnecessary Ducktape code use so in future, if need arise, rip this code to separate util more easily.